internal/resources/akscluster/resource_akscluster.go

```diff
@@ -0,0 +1,235 @@
+/*
+Copyright 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package akscluster
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/pkg/errors"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/client"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/akscluster"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	models "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/akscluster"
+)
+
+func ResourceTMCAKSCluster() *schema.Resource {
+	return &schema.Resource{
+		Schema:        ClusterSchema,
+		CreateContext: resourceClusterCreate,
+		ReadContext:   resourceClusterRead,
+		UpdateContext: resourceClusterInPlaceUpdate,
+		DeleteContext: resourceClusterDelete,
+		Description:   "Tanzu Mission Control AKS Cluster Resource",
+	}
+}
+
+// resourceClusterCreate will create an AKS cluster and any assigned nodepool.
+func resourceClusterCreate(ctx context.Context, data *schema.ResourceData, config any) diag.Diagnostics {
+	tc, ok := config.(authctx.TanzuContext)
+	if !ok {
+		return diag.Errorf("error while retrieving Tanzu auth config")
+	}
+
+	nodepools := ConstructNodepools(data)
+	if err := validate(nodepools); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := createOrUpdateCluster(data, tc.TMCConnection.AKSClusterResourceService); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := createNodepools(ctx, nodepools, tc.TMCConnection.AKSNodePoolResourceService); err != nil {
+		return diag.FromErr(err)
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, getTimeOut(data))
+	defer cancel()
+
+	if err := pollUntilReady(ctx, data, tc.TMCConnection, getPollInterval(ctx)); err != nil {
```

Comment:
True it will try to recreate the resource, but the logic above is allows use the existing resource in the event of an already exists conflict mirroring the EKS behavior.